src/core/dfn.js

```diff
@@ -86,3 +83,26 @@ function processAsInternalSlot(title, dfn) {
     showError(msg, name, { hint, elements: [dfn] });
   }
 }
+
+function addContractDefaults() {
+  // Find all dfns that don't have a type and default them to "dfn".
+  /** @type NodeListOf<HTMLElement> */
+  const dfnsWithNoType = document.querySelectorAll(
+    "dfn:is([data-dfn-type=''],:not([data-dfn-type]))"
+  );
+  for (const dfn of dfnsWithNoType) {
+    dfn.dataset.dfnType = "dfn";
+  }
+
+  // Per "the contract", export all definitions, except where:
+  //  - Explicitly marked with data-noexport.
+  //  - The type is "dfn" and not explicitly marked for export (i.e., just a regular definition).
+  //  - definitions was included via (legacy) data-cite="foo#bar".
+  /** @type NodeListOf<HTMLElement> */
+  const exportableDfns = document.querySelectorAll(
+    "dfn:not([data-noexport], [data-export], [data-dfn-type='dfn'], [data-cite])"
```

Comment:
@sidvishnoi, this is better :) Didn't realize I could just combine :not() like this instead of stacking them